src/utils/policy-utils.ts

```
@@ -0,0 +1,233 @@
+                await cleanExit(1, logger);
+            } else if(summaries.length > 1) {
+                logger.error(`More than 1 ${type} with name: ${resourceNameOrId}. Must use ${type} id instead.`);
+                await cleanExit(1, logger);
+            } else {
+                // If the resource exists
+                resource.id = summaries[0].id;
+            }
+        }
+        resources.push(resource);
+    }
+    return resources;
+}
+
+export async function getVirtualTargetsByNameOrId(targetNames: string[], dbHttpService: DbTargetService, webHttpService: WebTargetService, logger: Logger) {
+    // For each target, if it exists, grab the DbTargetSummary or WebTargetSummary and create a new Target object
+    // id = DbTargetSummary.id | WebTargetSummary.id, type = TargetType.Db | TargetType.Web
+    const allDbTargets = await dbHttpService.ListDbTargets();
+    const allWebTargets = await webHttpService.ListWebTargets();
+    return parseTwoTargetsTypesByNameOrId(targetNames, allDbTargets, allWebTargets, true, logger);;
+}
+
+export async function getBaseDacTargetsByNameOrId(targetNames: string[], baseTargetHttpService: BzeroTargetHttpService, dynamicAccessHttpService: DynamicAccessConfigHttpService, logger: Logger) {
+    // For each target, if it exists, grab the BzeroAgentSummary or DynamicAccessConfigSummary and create a new Target object
+    // id = BzeroAgentSummary.id | DynamicAccessConfigSummary.id, type = TargetType.Bzero | TargetType.DynamicAccessConfig
+    const allBaseTargets = await baseTargetHttpService.ListBzeroTargets();
+    const allDacTargets = await dynamicAccessHttpService.ListDynamicAccessConfigs();
+    return parseTwoTargetsTypesByNameOrId(targetNames, allBaseTargets, allDacTargets, false, logger);
+}
+
+async function parseTwoTargetsTypesByNameOrId(targetNames: string[], allTargetsType1: any[], allTargetsType2: any[], isVirtual: boolean, logger: Logger) {
```

For this i would suggest reusing the same function `parseSingleResourceTypeByNameOrId` but instead reduce all of the targets to a common interface (can use the existing `Target` interface). Then we just need to handle the case where a single target name maps to multiple targets. I think in this case we should just throw an error that says "more than 1 target exists with name <name> please specify the target using its uuid instead".